examples/image-compatibility.yaml

```diff
@@ -0,0 +1,49 @@
+version: v1alpha1
+compatibilities:
+- tag: "prefered"
+  weight: 10
+  description: "Prefered node configuration"
+  rules:
+  - name: "kernel and cpu"
+    vars:
+      my-prefered-kernel-and-cpu: "true"
+    matchFeatures:
+    - feature: kernel.loadedmodule
+      matchExpressions:
+        vfio-pci: {op: Exists}
+    - feature: cpu.model
+      matchExpressions:
+        vendor_id: {op: In, value: ["Intel", "AMD"]}
+  - name: "one of available nics"
+    vars:
+      required-nic: "true"
+    matchAny:
+    - matchFeatures:
+      - feature: pci.device
+        matchExpressions:
+          vendor: {op: In, value: ["0eee"]}
+          class: {op: In, value: ["0200"]}
+    - matchFeatures:
+      - feature: pci.device
+        matchExpressions:
+          vendor: {op: In, value: ["0fff"]}
+          class: {op: In, value: ["0200"]}
+  - name: "prefered requirements"
+    labels:
+      my-prefered-img-requirements: "true"
+    matchFeatures:
+    - feature: rule.matched
+      matchExpressions:
+        my-prefered-kernel-and-cpu: {op: IsTrue}
+        my-prefered-img-requirements: {op: IsTrue}
```

Comment:
reference to itself ? seems it's a self-loop reference. little complex to figure out the real use case.

maybe put the advanced back reference in another example

Reply:
Ok I will put it to another example. The backreference feature has been described here https://kubernetes-sigs.github.io/node-feature-discovery/v0.16/usage/customization-guide.html#backreferences